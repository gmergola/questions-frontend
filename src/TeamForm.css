.TeamForm-container {
  margin-left: 150px;
  margin-bottom: 40px;
  padding: 20px;
  max-width: 400px;
  border-radius: 15px;
  background-color: rgb(253, 179, 42);
  color: black;
}

.TeamForm-container:after {
  content: "";
  display: block;
  clear: both;
}

.TeamForm-team1 {
  font-size: 15px;
  float: left;
}

.TeamForm-team2 {
  font-size: 15px;
  float: right;
}

.TeamForm-team1-total {
  float: left;
}

.TeamForm-team2-total {
  float: right;
}

.TeamForm-save-button {
  margin-left: 20%;
  border-radius: 20px;
  width: 100px;
  height: 30px;
  border: none;
}

.TeamForm-save-button:hover {
  opacity: 0.5;
}

.TeamForm-team-div {
  margin-top: 60px;
  margin-bottom: 2px;
  font-size: 40px;
}

@media only screen and (max-width: 900px){
  .TeamForm-container {
    max-width: 300px;
  }

  .TeamForm-save-button {
    margin-left: 6%;
  }

  .TeamForm-team-div {
    font-size: 35px;
  }
}