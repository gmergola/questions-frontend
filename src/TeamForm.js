import React, { useState, useEffect } from 'react';
import './TeamForm.css';

/**TeamForm: a component that allows the user to determine which team is currently guessing an answer
 * allocates points to the current team playing
 */
function TeamForm({ currentVote, setIncorrectXs }) {
  const [team, setTeam] = useState('team1');
  const [team1Total, setTeam1Total] = useState(0);
  const [team2Total, setTeam2Total] = useState(0);

  function handleChange(evt) {
    setTeam(evt.target.value);
    setIncorrectXs([]);
  }

  function handleSubmit(evt) {
    evt.preventDefault();
  }

  /**changeTotals: adds to the total of the current team playing when a correct answer is guessed */
  function changeTotals() {
    if (currentVote !== undefined) {
      if (team === 'team1') {
        setTeam1Total((total) => total + currentVote);
      }
      if (team === 'team2') {
        setTeam2Total((total) => total + currentVote);
      }
    }
  }

  /**When currentVote state changes call changeTotals */
  useEffect(changeTotals, [currentVote]);

  return (
    <div className="TeamForm-container">
      <form onSubmit={handleSubmit}>
        <div className="form-check TeamForm-team1">
          <input
            onChange={handleChange}
            className="form-check-input"
            type="radio"
            name="teamRadios"
            id="teamRadios1"
            value="team1"
            checked={team === "team1"}></input>
          <label htmlFor="teamRadios1"><b>Team 1</b></label>
        </div>
        <div className="form-check TeamForm-team2">
          <input
            onChange={handleChange}
            className="form-check-input"
            type="radio"
            name="teamRadios"
            id="teamRadios2"
            value="team2"
            checked={team === "team2"}></input>
          <label htmlFor="teamRadios2"><b>Team 2</b></label>
        </div>
        <button className="TeamForm-save-button"><b>Save</b></button>
      </form>
      <h5 className="TeamForm-team1-total">Team 1 total: {team1Total}</h5>
      <h5 className="TeamForm-team2-total" >Team 2 total: {team2Total}</h5>
      <div className="TeamForm-team-div"><b>{`${team[0].toLocaleUpperCase()}${team.slice(1, team.length-1)} ${team[team.length-1]}`}</b></div>
    </div>
  )
}

export default TeamForm;