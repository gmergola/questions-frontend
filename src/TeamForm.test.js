import React from "react";
import { render } from "@testing-library/react";
import TeamForm from "./TeamForm";

it("renders without crashing", () => {
  render(<TeamForm />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<TeamForm />);
  expect(asFragment()).toMatchSnapshot();
});