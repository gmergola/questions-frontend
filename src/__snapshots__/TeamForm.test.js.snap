// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TeamForm-container"
  >
    <form>
      <div
        class="form-check TeamForm-team1"
      >
        <input
          checked=""
          class="form-check-input"
          id="teamRadios1"
          name="teamRadios"
          type="radio"
          value="team1"
        />
        <label
          for="teamRadios1"
        >
          <b>
            Team 1
          </b>
        </label>
      </div>
      <div
        class="form-check TeamForm-team2"
      >
        <input
          class="form-check-input"
          id="teamRadios2"
          name="teamRadios"
          type="radio"
          value="team2"
        />
        <label
          for="teamRadios2"
        >
          <b>
            Team 2
          </b>
        </label>
      </div>
      <button
        class="TeamForm-save-button"
      >
        <b>
          Save
        </b>
      </button>
    </form>
    <h5
      class="TeamForm-team1-total"
    >
      Team 1 total: 0
    </h5>
    <h5
      class="TeamForm-team2-total"
    >
      Team 2 total: 0
    </h5>
    <div
      class="TeamForm-team-div"
    >
      <b>
        Team 1
      </b>
    </div>
  </div>
</DocumentFragment>
`;
